pkg/stanza/fileconsumer/internal/archive/archive_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package archive_test // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/archive_test"
+
+import (
+	"context"
+	"fmt"
+	"math/rand/v2"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/archive"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fileset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fingerprint"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/testutil"
+)
+
+func TestArchiveCRUD(t *testing.T) {
+	// pollsToArchiveMatrix contains different polls_to_archive settings to test
+	pollsToArchiveMatrix := []int{10, 20, 50, 100, 200}
+
+	for _, p := range pollsToArchiveMatrix {
+		t.Run(fmt.Sprintf("pollsToArchive:%d", p), func(t *testing.T) { testArchive(t, p) })
+	}
+
+	t.Run("nopArchive", func(t *testing.T) {
+		testArchiveNop(t)
+	})
+}
+
+func testArchive(t *testing.T, pollsToArchive int) {
+	persister := testutil.NewUnscopedMockPersister()
+	a := archive.New(context.Background(), zap.L(), pollsToArchive, persister)
+
+	m := make(map[int]*fingerprint.Fingerprint)
+
+	// rolledOverFps contains the fingerprints that were previously a part of archive, but are now rolled over i.e. removed.
+	// archive should no longer contain such fingerprints
+	rolledOverFps := make([]*fingerprint.Fingerprint, 0)
+
+	for i := 0; i < 50; i++ {
+		fp := fingerprint.New([]byte(createRandomString(100)))
+
+		if oldFp, isRollover := m[i%pollsToArchive]; isRollover {
```

Comment:
What is the significance of `i`? What is the significance of `i%pollsToArchive`?